### ISSUE TITLE
CustomData can't be preloaded with Collections

### ISSUE THREAD

**Issue by x0r04rg** [ezEngine/ezEngine#1235]
**Bug description:**
CustomData assets referenced by a Collection do not get preloaded when they should.

**How to Reproduce:**
STR:
1. Add a CustomData entry to a Collection.
2. Load the collection resource.
3. Call `ezCollectionResource::PreloadResources` method on it.

Expected:
CustomData gets preloaded.

Actual:
CustomData doesn't get preloaded. The error is printed to the log: `There was no valid RTTI available for assets with type name 'CustomData'. Could not pre-load resource '{ 471761a4-3e8f-41ca-928a-4bf1d67b9e08 }'. Did you forget to register the resource type with the ezResourceManager?`

**Additional comment:**
It happens because "CustomData" asset type is not registered in `ezGameApplication::Init_ConfigureAssetManagement`. But it cannot be easily added there, because there is no universal/typeless resource type for a custom data - `ezCustomDataResource` is a template.

**Comment by jankrassnigg** [comment]
Yes, this is tricky to solve. There are a couple of ways one can try to work around the issue, but they are all ugly. I need some time to think about this.

**Comment by C-Core** [comment]
Atm we use the startup system for other resource types that have the same problem, like e.g. the ezVisualScriptClassResource.
